File: Source/STSL.SmartLocker.Utils.WebClient/src/app/shared/models/tenant.ts
import { IUsesGuidId } from "./common";

export type Base64EncodedImage = string;

export interface ITenantDTO extends IUsesGuidId {
  name: string;
  logo?: Base64EncodedImage;
  logoMimeType?: string;
  cardHolderAliasSingular?: string;
  cardHolderAliasPlural?: string;
  cardHolderUniqueIdentifierAlias?: string;
  helpPortalUrl?: string;
  allowLockUpdates?: boolean;
}

export interface ICreateTenantDTO {
  name: string;
  logo?: Base64EncodedImage;
  logoMimeType?: string;
  cardHolderAliasSingular?: string;
  cardHolderAliasPlural?: string;
  cardHolderUniqueIdentifierAlias?: string;
  helpPortalUrl?: string;
}

export interface IUpdateTenantDTO {
  name: string;
  logo?: Base64EncodedImage;
  logoMimeType?: string;
  cardHolderAliasSingular?: string;
  cardHolderAliasPlural?: string;
  cardHolderUniqueIdentifierAlias?: string;
  helpPortalUrl?: string;
}

File: Source/STSL.SmartLocker.Utils.Domain/Tenant.cs
namespace STSL.SmartLocker.Utils.Domain;

public class Tenant : EntityBase
{
    public required string Name { get; set; }
    public bool AllowLockUpdates { get; set; } = false;
    public string CardHolderAliasSingular { get; set; } = "Card Holder";
    public string CardHolderAliasPlural { get; set; } = "Card Holders";
    public string CardHolderUniqueIdentifierAlias { get; set; } = "Unique Identifier";
    //public required string LockerServiceTagBaseUrl { get; set; }
    public string? HelpPortalUrl { get; set; }
    public byte[]? Logo { get; set; }
    public string? LogoMimeType { get; set; }
}
